File: Esport-Backend/Common/Regions.cs
namespace TForge.Common
{
    /// <summary>
    /// Канонічні кіберспортивні регіони. Той самий поділ, що у Games та
    /// Countries: тут значення, а підписи — на клієнті
    /// (Esport-Frontend/src/constants/regions.ts).
    ///
    /// Вільний текст давав «Europe», «EU» і «Європа» як три різні регіони, за
    /// якими не згрупувати нічого. Значення збігаються з тими, що вже лежать у
    /// базі, тож наявні команди лишаються валідними без міграції.
    /// </summary>
    public static class Regions
    {
        public const string Europe = "Europe";
        public const string NorthAmerica = "North America";
        public const string SouthAmerica = "South America";
        public const string Cis = "CIS";
        public const string Asia = "Asia";
        public const string Oceania = "Oceania";
        public const string MiddleEast = "Middle East";
        public const string Africa = "Africa";

        public static readonly string[] All =
        {
            Europe, NorthAmerica, SouthAmerica, Cis, Asia, Oceania, MiddleEast, Africa
        };

        public static bool IsValid(string? region) => region != null && All.Contains(region);
    }
}

File: Esport-Frontend/src/constants/regions.ts
/**
 * Дзеркало Esport-Backend/Common/Regions.cs.
 *
 * Значення (ключі) — це те, що зберігає база й перевіряє валідатор.
 * Підписи — те, що бачить користувач. Той самий поділ, що у games.ts та
 * countries.ts: змінюючи список тут, змініть його і на бекенді.
 */
export const REGIONS = [
  "Europe",
  "North America",
  "South America",
  "CIS",
  "Asia",
  "Oceania",
  "Middle East",
  "Africa"
] as const;

export type Region = (typeof REGIONS)[number];

export const REGION_LABELS: Record<Region, string> = {
  Europe: "Європа",
  "North America": "Північна Америка",
  "South America": "Південна Америка",
  CIS: "СНД",
  Asia: "Азія",
  Oceania: "Океанія",
  "Middle East": "Близький Схід",
  Africa: "Африка"
};

/** Підпис для значення з бекенда, яке може бути й невідомим (старі дані). */
export const regionLabel = (region: string): string =>
  (REGION_LABELS as Record<string, string>)[region] ?? region;
